### PR TITLE
solana-ibc: use Borsh for consensus state serialisation

### PR DESCRIPTION
**Pull request by mina86**
Closes: https://github.com/ComposableFi/emulated-light-client/issues/46

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -30,8 +30,8 @@ borsh = { version = "0.10.3", default-features = false }
 bytemuck = { version = "1.14", default-features = false }
 derive_more = "0.99.17"
 hex-literal = "0.4.1"
-ibc = { version = "0.47.0", default-features = false, features = ["serde", "borsh"] }
-ibc-proto = { version = "0.37.1", default-features = false, features = ["serde"] }
+ibc = { version = "0.47.0", default-features = false, features = ["borsh"] }
```
Comment (dhruvja) [inline Cargo.toml:33]: I think the `ibc::applications::transfer` requires the `serde` feature. So we have to keep the `serde` feature

Comment (mina86) [reply Cargo.toml]: Yes, you’re right. It’s weird that `cargo test` run didn’t fail compiling the code.

Comment (mina86) [reply Cargo.toml]: OK, actually, it’s probably because some dev dependency enables serde feature.

Comment (dhruvja) [reply Cargo.toml]: oh damn, got it

**Review by dhruvja — APPROVED** [review]

**Pull request merged**